ros2/README.md

```diff
@@ -0,0 +1,185 @@
+
+### Quality of Service Object
+This object contains ROS 2 specific information about the Quality of Service policies.
+More information here: https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-policies
+
+Field Name | Type | ROS 2 Versions | Description
+---|:---:|:---:|---|
+`reliability` | string | all | One of `best_effort` or `reliable`.  More information here: [ROS 2 QoS](https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-policies)
+`history` | string | all | One of `keep_last`, `keep_all` or `unknown`. More information here: [ROS 2 QoS](https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-policies)
+`durability` | string | all | One of `transient_local` or `volatile`. More information here: [ROS 2 QoS](https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-policies)
+`lifespan` | integer | all | The maximum amount of time between the publishing and the reception of a message without the message being considered stale or expired. `-1` means infinite. 
+`deadline` | integer | all | The expected maximum amount of time between subsequent messages being published to a topic. `-1` means infinite.
+`liveliness` | string | all | One of `automatic`or `manual`. More information here: [ROS 2 QoS](https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-policies)
+`leaseDuration` | integer | all | the maximum period of time a publisher has to indicate that it is alive before the system considers it to have lost liveliness. `-1` means infinite.
+### Examples
+
+ROS 2 subscriber example:
+
+```yaml
+  receiveCmdVel:
+    action: receive
+    channel:
+      $ref: "#/channels/CmdVel"
+    bindings:
+      ros2:
+        role: subscriber
+        node: /turtlesim
+          qosPolicies:
+            history: unknown
+            reliability: reliable
+            durability: volatile
+            lifespan: -1
+            deadline: -1
+            liveliness: automatic
+            leaseDuration: -1
+```
+ROS 2 action Server example:
+
+```yaml
+  receiveRotateAbsolute:
+    action: receive
+    channel:
+      $ref: "#/channels/RotateAbsoluteRequest"
+    reply:
+      channel:
+        $ref: "#/channels/RotateAbsoluteReply"
+    bindings:
+      ros2:
+        role: action_server
+        node: /turtlesim
+```
+
+
+<a name="message"></a>
+
+## Message Binding Object
+
+ROS 2 message types (defined in .msg/.srv/.aciton files) are mapped to AsyncAPI message payloads.
+
+This object DOES NOT contain any ROS 2 specific properties.
+
+```yaml
+Vector3Msg:
+  type: object
+  properties: 
+    x:
+      type: number
+      format: double
+    y:
+      type: number
+      format: double
+    z:
+      type: number
+      format: double
+```
+
+ROS 2  Type | AsyncAPI Type | AsyncAPI Format | 
```

Comment:
```suggestion
ROS 2 Type | AsyncAPI Type | AsyncAPI Format | 
```

Reply:
What's this table about? When should it be used and what for?

Reply:
Sure, I'm all in with having it, I just think we have to explain what is for. Right now it's just there without any further explanation.

Reply:
Sure, I'm all in with having it, I just think we have to explain what is for. Right now it's just there without any further explanation.

Reply:
I do believe this belongs here, agreed with adding explanation. Suggest including a link to the [design document](https://design.ros2.org/articles/legacy_interface_definition.html)

Reply:
We included the design document suggested by @Achllle .

But @fmvilas question is not answered:
>  explain what [this table] is for

I think such an information could be needed to properly integrate/visualize different bindings in one tool. That is why you have your [AsyncAPI message types and formats](https://www.asyncapi.com/docs/reference/specification/v3.0.0#dataTypeFormat:~:text=The%20formats%20defined%20by%20the%20AsyncAPI%20Specification%20are%3A) for.

Right now, this is used as our Rosetta Stone table for the siemens PoC AsyncAPI generator reading ros message file definitions and producing respecting AsyncAPI files.

I hope that the current wording reflects this already. If not, we can make it more distinctive.

Reply:
I think this conversation can be marked as resolved already, right? Unless you want to provide more information.